site/en/blog/videong/index.md

```diff
@@ -0,0 +1,247 @@
+macOS has
+[CoreAnimation Layers](https://developer.apple.com/documentation/quartzcore/calayer) and
+[VideoToolbox](https://developer.apple.com/documentation/videotoolbox),
+Android has
+[SurfaceView](https://developer.android.com/reference/android/view/SurfaceView) and
+[MediaCodec](https://developer.android.com/reference/android/media/MediaCodec), and Linux has
+[VASurfaces](https://source.chromium.org/chromium/chromium/src/+/main:media/gpu/vaapi/va_surface.h;l=20;drc=e0b362edd9b49143b89fc76c4a31dd5603b6fbd0) and
+[VA-API](https://01.org/linuxmedia/vaapi).
+Chromium's abstractions for these concepts are handled by
+the
+[OverlayProcessor](https://source.chromium.org/chromium/chromium/src/+/main:components/viz/service/display/overlay_processor_interface.h;drc=0deb8efb4a2bbf029b1990ca5e1dc96037ac25f8) and
+[mojo::VideoDecoder](https://source.chromium.org/chromium/chromium/src/+/main:media/mojo/mojom/video_decoder.mojom;drc=4041de9d04f91b51497e7769608391a8e8bedf5e) interfaces respectively.
+
+In some cases it's possible for these buffers to be mappable into system memory,
+so they don't even need to be opaque and don't consume any bandwidth until accessed—Chromium calls these
+[GpuMemoryBuffers](https://docs.google.com/document/d/1SaTYTBvHWWDKA3MPJPpQ-79RNgdS4Xu4g3KiD39VQjU/preview).
+On Windows these are backed by
+[DXGI buffers](https://docs.microsoft.com/en-us/windows/win32/direct3ddxgi/d3d10-graphics-programming-guide-dxgi),
+on macOS
+[IOSurfaces](https://developer.apple.com/documentation/iosurface),
+on Android
+[AHardwareBuffers](https://developer.android.com/ndk/reference/group/a-hardware-buffer), and on Linux
+[DMA buffers](https://01.org/linuxgraphics/gfx-docs/drm/driver-api/dma-buf.html#buffer-sharing-and-synchronization).
+While video playback generally doesn't need this access,
+these buffers are important for video capture to ensure minimal bandwidth between the capture device and eventual encoders.
+
+{% Img src="image/ZDZVuXt6QqfXtxkpXcPGfnygYjd2/oOERhv9vDRkFKnKSjFOf.png",
+alt="Diagram of the buffers mentioned in the preceding text.",
+width="580", height="420" %}
+
+Since the GPU is often responsible for both decoding and displaying,
+the use of these (also often) opaque buffers ensures that high bandwidth video data never actually leaves the GPU.
+As we discussed earlier,
+keeping data on the GPU is incredibly important for efficiency;
+especially at high resolutions and frame rates.
+
+The more we can take advantage of OS primitives like overlays and GPU buffers,
+the less bandwidth is spent shuffling video bytes around unnecessarily.
+Keeping everything in one place from decoding all the way to rendering can lead to incredible power efficiency.
+For example, when Chromium
+[enabled overlays](https://bugs.chromium.org/p/chromium/issues/detail?id=594449) on macOS,
+power consumption during video playback was halved!
+
```

Comment:
```suggestion
On Windows and ChromeOS, hardware overlays (through [swap chains](https://docs.microsoft.com/en-us/windows/win32/direct3d9/what-is-a-swap-chain-) on Windows and an equivalent concept on ChromeOS) can be used on some devices to bypass Chrome’s compositor and the OS compositor, leading to much less overhead from copying and drawing. This can save up to 50% of battery drain when playing videos. 
```